packages/mui-material-next/src/IconButton/IconButton.types.ts

```diff
@@ -58,12 +58,14 @@ export interface IconButtonOwnProps {
   sx?: SxProps<Theme>;
 }
 
+export interface IconButtonOwnerState extends IconButtonProps {}
```

Comment:
```suggestion
export interface IconButtonOwnerState extends PartiallyRequired<IconButtonProps, 'edge' | 'color' | 'size'> {}
```
We could use [`PartiallyRequired`](https://github.com/mui/material-ui/pull/40227) for the ones with default values